src/fastnetmon.cpp

```diff
@@ -3581,20 +3581,29 @@ void produce_dpi_dump_for_pcap_dump(std::string pcap_file_path, std::stringstrea
     // Buffer for packets
     char packet_buffer[pcap_header.snaplen];
 
-    unsigned int total_packets_number = 0;
-
+    uint64_t total_packets_number = 0;
     uint64_t dns_amplification_packets = 0;
     uint64_t ntp_amplification_packets = 0;
     uint64_t ssdp_amplification_packets = 0;
     uint64_t snmp_amplification_packets = 0;
 
+    struct ndpi_id_struct *src = NULL;
```

Comment:
Yep, it will work per packet basis here but in common case I prefer to keep
one instance per analyzed file.

On Wednesday, 10 August 2016, dalibort <notifications@github.com> wrote:

> In src/fastnetmon.cpp
> <https://github.com/pavel-odintsov/fastnetmon/pull/540#discussion_r74325954>
> :
>
> >      uint64_t dns_amplification_packets = 0;
> >      uint64_t ntp_amplification_packets = 0;
> >      uint64_t ssdp_amplification_packets = 0;
> >      uint64_t snmp_amplification_packets = 0;
> >
> > +    struct ndpi_id_struct *src = NULL;
>
> Hi,
> if you are talking about src/dst/flow ndpi vars - your original code
> allocates and zeroes them per packet too. I am not sure what full tracking
> is for in this case. I think here we don;t need to analyze tcp and flows.
> In this time the attack was already analyzed and detected. This code only
> tries to match the packets with well known amplification attacks.
> Maybe I am missing something?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/pavel-odintsov/fastnetmon/pull/540/files/fcfcc0e6744db22c778cf51341966700c727ffc1#r74325954>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACnfZrLveF6Fkpbwlwt4goYqndb4F8nCks5qejc8gaJpZM4JfEE5>
> .
>


-- 
Sincerely yours, Pavel Odintsov